core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.spec.ts

```diff
@@ -13,98 +13,84 @@
-  var TopicsAndSkillsDashboardBackendApiService = null;
-  var $httpBackend = null;
-  var UrlInterpolationService = null;
   var SAMPLE_TOPIC_ID = 'hyuy4GUlvTqJ';
 
-  var sampleDataResults = {
-    topic_summary_dicts: [{
-      id: SAMPLE_TOPIC_ID,
-      name: 'Sample Name',
-      language_code: 'en',
-      version: 1,
-      canonical_story_count: 3,
-      additional_story_count: 0,
-      uncategorized_skill_count: 3,
-      subtopic_count: 3,
-      topic_model_created_on: 1466178691847.67,
-      topic_model_last_updated: 1466178759209.839
-    }],
-    skill_summary_dicts: []
-  };
 
   var TOPICS_AND_SKILLS_DASHBOARD_DATA_URL =
-    '/topics_and_skills_dashboard/data';
-  var ERROR_STATUS_CODE = 500;
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(
-    angular.mock.module('oppia', TranslatorProviderForTests));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector) {
-    TopicsAndSkillsDashboardBackendApiService = $injector.get(
-      'TopicsAndSkillsDashboardBackendApiService');
-    UrlInterpolationService = $injector.get('UrlInterpolationService');
-    $httpBackend = $injector.get('$httpBackend');
-  }));
-
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
-  });
-
-  it('should successfully fetch topics and skills dashboard data from the ' +
+  '/topics_and_skills_dashboard/data';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+    });
+    httpTestingController = TestBed.get(HttpTestingController);
+    // eslint-disable-next-line max-len
+    TopicsAndSkillsDashboardBackendApiService = TestBed.get(TopicsAndSkillsDashboardBackendApiService);
+    var sampleDataResults = {
+      topic_summary_dicts: [{
+        id: SAMPLE_TOPIC_ID,
+        name: 'Sample Name',
+        language_code: 'en',
+        version: 1,
+        canonical_story_count: 3,
+        additional_story_count: 0,
+        uncategorized_skill_count: 3,
+        subtopic_count: 3,
+        topic_model_created_on: 1466178691847.67,
+        topic_model_last_updated: 1466178759209.839
+      }],
+      skill_summary_dicts: []
+    };
+
+    afterEach(() => {
+      httpTestingController.verify();
+    });
+    it('should use rejection handler if dashboard data backend request failed',
+      fakeAsync(()=>{
+        var successHandler = jasmine.createSpy('success');
+        var failHandler = jasmine.createSpy('fail');
+        TopicsAndSkillsDashboardBackendApiService.fetchDasboardData().then(
+          successHandler, failHandler);
+        var req = httpTestingController.expectOne(
+          //eslint-disable-next-line max-len
```

Comment:
Only disable eslint as a last resort, you can break this up after the comma to avoid the lint error.